libmscore/figuredbass.cpp

```diff
@@ -1770,6 +1773,256 @@ FiguredBass* Score::addFiguredBass()
+            Element *element = segment->element(track);
+            if ( element == 0)
+                  continue;
+            if ( element->type() != Element::Type::CHORD)
+                  continue;
+            {
+                  vector<int> harmonys;
+                  Chord *chord = static_cast<Chord*>(element);
+                  if ( calcFiguredBass(chord, harmonys) ) { // appends entries into harmonys corresponding musical harmony pitches
+                        for (int pitch : harmonys) {
+                              NoteVal *nval = new NoteVal(pitch);
+                              chord->score()->addNote(chord,*nval);
+                        }
+                  } else {
+                        //qDebug("no figured bass on chord on track %d", chord->track());
+                  }
+            }
+      }
+}
+      
+// populate the given harmonys vector with the pitches of the harmony notes implied by the figured base on the given chord
+bool FiguredBass::calcFiguredBass(Chord *chord, vector<int> & harmonys) {
+      Q_ASSERT(harmonys.size() == 0);
+      Segment *segment = chord->segment();
+      int octave = 0; // 0, or 12 if we want the harmony 1 octave up
+      FiguredBassItem::Modifier none = FiguredBassItem::Modifier::NONE;
+      if (find_if(segment->annotations().cbegin(),
+                  segment->annotations().cend(),
+                  [chord](Element *e) {
+                        return ( e->type() == Element::Type::FIGURED_BASS
+                                && e->track() == chord->track());
+                  }) == segment->annotations().cend())
+      {
+             return false; // no figured base to annotate
+      }
+      
+      if ( chord->notes().size() == 0 ) {
+            qDebug("no notes on chord");
+            return false; // no notes in the chord, not sure if this is possible?
+      }
+      if ( chord->notes().size() != 1 ) {
+            qDebug("measure #%d, cannot render Figured Base on a multi-note chord", segment->measure()->no());
+            return false;
+      }
+      Note *note = chord->notes()[0];
+
+      auto mod7 = [] (int n) {
+            n = n%7;
+            while ( n < 0)
+                  n += 7;
+            return n;
+      };
+      auto mod12 = [] (int n) {
+            n = n%12;
+            while ( n < 0)
+                  n += 12;
+            return n;
+      };
+      auto get_digit = [](int digit) {
+            // a modifier with no digit always references the 3rd
+            return (digit == FBIDigitNone ) ? 3 : digit;
+      };
+      auto render_harmony = [&] (FiguredBassItem::Modifier prefix, int relpitch, FiguredBassItem::Modifier suffix) {
+            relpitch = get_digit(relpitch);
+            if ( prefix != none && suffix != none) {
+                  qDebug("(1242) measure #%d, cannot render FiguredBassModifier with two modifiers", segment->measure()->no());
+                  return;
+            }
```

Comment:
Possibly useful, but the controls on figured bass input already reject such a case. An example of why it would be preferable not to allow public (write) access to FiguredBassItem list.